tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/DockerTestFramework.cs

```diff
@@ -0,0 +1,337 @@
+            {
+            }
+
+            protected override async void RunTestCases(IEnumerable<IXunitTestCase> testCases, IMessageSink executionMessageSink, ITestFrameworkExecutionOptions executionOptions)
+            {
+                using (var assemblyRunner = new CustomAssemblyRunner(TestAssembly, testCases, DiagnosticMessageSink, executionMessageSink, executionOptions))
+                {
+                    await assemblyRunner.RunAsync();
+                }
+            }
+        }
+
+        private class CustomAssemblyRunner : XunitTestAssemblyRunner
+        {
+            public CustomAssemblyRunner(ITestAssembly testAssembly, IEnumerable<IXunitTestCase> testCases, IMessageSink diagnosticMessageSink, IMessageSink executionMessageSink, ITestFrameworkExecutionOptions executionOptions)
+                : base(testAssembly, testCases, diagnosticMessageSink, executionMessageSink, executionOptions)
+            {
+            }
+
+            protected override async Task<RunSummary> RunTestCollectionsAsync(IMessageBus messageBus, CancellationTokenSource cancellationTokenSource)
+            {
+                var collections = OrderTestCollections().Select(
+                    pair =>
+                    new
+                    {
+                        Collection = pair.Item1,
+                        TestCases = pair.Item2,
+                        DisableParallelization = IsParallelizationDisabled(pair.Item1)
+                    })
+                    .ToList();
+
+                TestcontainersSettings.ResourceReaperEnabled = false;
+
+                var summary = new RunSummary();
+
+                using var runner = new ConcurrentRunner();
+
+                var tasks = new List<Task<RunSummary>>();
+
+                var containerFixtures = collections
+                    .SelectMany(c => c.TestCases)
+                    .Select(t => t.Method.ToRuntimeMethod().DeclaringType)
+                    .Where(t => t.GetInterfaces().Any(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IClassFixture<>)))
+                    .ToList();
+
+                foreach (var type in containerFixtures)
+                {
+                    // Retrieve all the types of container fixtures
+                    var fixtureTypes = type.GetInterfaces()
+                        .Where(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IClassFixture<>))
+                        .Select(i => i.GetGenericArguments()[0])
+                        .Where(t => typeof(ContainerFixture).IsAssignableFrom(t))
+                        .ToList();
+
+                    foreach (var fixtureType in fixtureTypes)
+                    {
+                        try
+                        {
+                            var fixture = (ContainerFixture)Activator.CreateInstance(fixtureType);
+                            await fixture!.InitializeAsync();
```

Comment:
Out of interest, why do we need this? 🤔 Doesn't xunit automatically call this for fixtures?

Reply:
We don't _need_ this. But I prefer to give the responsibility of starting the containers to the executor, at a well-defined time, rather than letting xunit do its thing.